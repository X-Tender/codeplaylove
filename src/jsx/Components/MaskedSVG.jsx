import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	text: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
	color: PropTypes.string.isRequired,
};

const MaskedSVG = ({ text, type, color }) => (
	<svg className="title__svg" height="100%" width="100%">
		<rect
			className="knockout-text-bg"
			fill={color}
			fillOpacity="0.4"
			height="100%"
			mask={`url(#knockout-text-${type})`}
			width="100%"
			x="0"
			y="0"
		/>

		<mask className="title__mask" id={`knockout-text-${type}`}>
			<rect fill="#FFF" height="100%" width="100%" x="0" y="0" />
			<text
				alignmentBaseline="central"
				dominantBaseline="central"
				fill="#000"
				textAnchor="middle"
				x="50%"
				y="50%"
			>
				{text}
			</text>
		</mask>
	</svg>
);

MaskedSVG.propTypes = propTypes;

export default MaskedSVG;
