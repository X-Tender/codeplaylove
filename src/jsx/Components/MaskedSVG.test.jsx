import React from 'react';
import { shallow } from 'enzyme';
import MaskedSVG from './MaskedSVG';

describe('MaskedSVG', () => {
	const text = 'CODE';
	const type = 'code';
	const color = '#ff0000';

	const comp = shallow(<MaskedSVG color={color} text={text} type={type} />);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders the correct props', () => {
		expect(comp.find('rect.knockout-text-bg').prop('fill')).toEqual(color);
		expect(comp.find('rect.knockout-text-bg').prop('mask')).toEqual(`url(#knockout-text-${type})`);
		expect(comp.find('mask.title__mask').prop('id')).toEqual(`knockout-text-${type}`);
		expect(comp.find('mask text').text()).toEqual(text);
	});
});
