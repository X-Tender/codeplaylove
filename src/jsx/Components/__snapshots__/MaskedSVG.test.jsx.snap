// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaskedSVG renders correctly 1`] = `
<svg
  className="title__svg"
  height="100%"
  width="100%"
>
  <rect
    className="knockout-text-bg"
    fill="#ff0000"
    fillOpacity="0.4"
    height="100%"
    mask="url(#knockout-text-code)"
    width="100%"
    x="0"
    y="0"
  />
  <mask
    className="title__mask"
    id="knockout-text-code"
  >
    <rect
      fill="#FFF"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
    <text
      alignmentBaseline="central"
      dominantBaseline="central"
      fill="#000"
      textAnchor="middle"
      x="50%"
      y="50%"
    >
      CODE
    </text>
  </mask>
</svg>
`;
